integrations/tensorflow/compiler/dialect/tf_tensorlist/ir/tf_tensorlist_ops.td

```diff
@@ -138,4 +138,45 @@ def TfTensorList_Stack : TfTensorList_Op<"Stack"> {
   );
 }
 
+def TfTensorList_Concat : TfTensorList_Op<"Concat", [NoSideEffect]> {
+  let summary = "Concatenates all tensors in the list along the first dimension.";
+
+  let description = [{
+    Requires that all tensors have the same shape along dimensions 1+.
+
+    list: the input list
+    tensor: the concatenated result
+  }];
+
+  let arguments = (ins
+    TfTensorList_TensorList:$list
+  );
+
+  let results = (outs
+    TF_Tensor:$tensor
+  );
+}
+
+// TODO(suderman): This op purely exists to be optimized away right now.
```

Comment:
This is used by the gradient code. Maybe rewrite this as "if gradients are not being calculated, the value will be DCE'd"

Reply:
Done.